Google.PowerShell/Common/AuthenticateWithSdkCredentialsExecutor.cs

```diff
@@ -125,13 +97,20 @@ public void Initialize(ConfigurableHttpClient httpClient)
 
         public virtual async Task<string> GetAccessTokenForRequestAsync(string authUri = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (Token.IsExpired(flow.Clock))
+            if (s_token == null)
+            {
+                await RefreshTokenAsync(cancellationToken).ConfigureAwait(false);
+                return s_token.AccessToken;
+            }
+
+            if (s_token.IsExpiredOrInvalid())
```

Comment:
Following up with a comment I made earlier, it makes sense that the `AuthenticateWithSdkCredentialsExecutor` class knows about the Cloud SDK and so we can put the "has the active user changed" logic here. Perhaps:

````
string currentCloudSdkUser = CloudSDKConfig.GetProperty("user");
if (s_token.IsExpired || (s_token.User != currentCloudSdkUser)) {
    // refresh the token...
}
````